(Chap_ExceptIO)=
# Exception Handling and File Input/Output

Python stops the computation process when it encounters an error. Sometimes you want to describe certain errors with descriptive error messages. And sometimes you want your code to move past errors while saving them and including descriptive error messages.

```{exercise-start}
:label: ExerExceptionIO
```
Read the BYU ACME "[Exceptions and file input/output](https://acme.byu.edu/0000017c-ccfe-da17-a5fd-cdfeff8a0000/acmefiles-08-exceptions-fileio-2021-pdf)" lab and complete Problems 1 through 4 in the lab. {cite}`BYUACME_ExceptIO`
```{exercise-end}
```
